app/src/main/java/com/mstudio/superheroarch/presentation/DetailsViewModel.kt

```
@@ -1,40 +1,84 @@
+                view.showError("Failed to load TMDB episode rating: ${e.message}")
+            }
+        }
+    }
+
+    fun extractSeasonAndEpisode(firstEpisode: String): Pair<Int, Int>? {
+        return if (firstEpisode.isNotEmpty()) {
+            try {
+                val season = firstEpisode.substring(1, 3).toInt()   // Extrae el número de temporada
+                val episode = firstEpisode.substring(4, 6).toInt()  // Extrae el número de episodio
+                Pair(season, episode)
+            } catch (e: Exception) {
```

detekt

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)